distribution/lib/Standard/Base/0.0.0-dev/src/Random.enso

```diff
@@ -1,99 +1,307 @@
+
+    ## GROUP Random
+       Returns a random permutation of the input `Vector`.
+
+       Arguments:
+       - vector: The `Vector` to permute.
+
+       > Example
+         Permute a vector.
+
+             import Standard.Base.Random.Random
+
+             permuted = Random.permute [1, 2, 3, 4]
+    permute : Vector Any -> Vector Any
+    permute (v:Vector) = Random.default_instance.permute v
 
     ## PRIVATE
-       Serialise to JS_Object
-    to_js_object : JS_Object
-    to_js_object self =
-        JS_Object.from_pairs [["type", "Random_Number_Generator"], ["constructor", "new"]]
+       Return the default instance.
+    default_instance : Random_Generator
+    default_instance = Random_Generator.Value Random_Utils.getRandomSingleton
 
+# PRIVATE
+type Random_Generator
     ## PRIVATE
-       Convert to a display representation of this Random_Number_Generator.
-    to_display_text : Text
-    to_display_text self = "Random_Number_Generator"
+       Create a new rng with the given seed.
+    new : Integer -> Random_Generator
+    new (seed:Integer) = Random_Generator.Value (Java_Random.new seed)
 
-## PRIVATE
-   Returns a new vector containing a random sample of the input vector, without
-   replacement.
+    ## PRIVATE
+    Value (java_random:Java_Random)
 
-   If the amount of elements to select is larger than the input vector size, it
-   returns a random permutation of the input vector.
-sample : Vector Any -> Integer -> Random_Number_Generator -> Vector Any
-sample vector k rng =
-    new_array = Random_Utils.sample vector k rng.java_random
-    Vector.from_polyglot_array new_array
+    integer : Integer -> Integer -> Integer ! Illegal_Argument
+    integer self (min:Integer) (max:Integer) =
+        range = max - min
+        if range >= Java_Integer.MAX_VALUE then Error.throw (Illegal_Argument.Error "Currently only integer ranges of up to 2^31-1 are supported.") else
+          min + (self.java_random.nextInt range)
 
-## PRIVATE
-   Returns `k` indices sampled from the range [0, n-1] without replacement.
+    float : Float -> Float -> Float
+    float self (min:Float=0.0) (max:Float=1.0) =
+        range = max - min
+        self.java_random.nextDouble * range + min
 
-   If `k >= n`, it will return a random permutation of the indices.
-random_indices : Integer -> Integer -> Random_Number_Generator -> Vector Integer
-random_indices n k rng =
-    array = Random_Utils.random_indices n k rng.java_random
-    Vector.from_polyglot_array array
+    boolean : Boolean
+    boolean self =
+        self.java_random.nextBoolean
+
+    gaussian : Float -> Float -> Float
+    gaussian self (standard_deviation:Float=1.0) (mean:Float=0.0) =
+        self.java_random.nextGaussian * standard_deviation + mean
+
+    indices : Integer -> Integer -> Vector Integer
+    indices self (n:Integer) (count:Integer=1) =
+        array = Random_Utils.random_indices n count self.java_random
+        Vector.from_polyglot_array array
```

Comment:
Btw. there is `Vector.collect` since #8035 to possibly simplify these kind of _"iterate and build vector"_ actions.

Reply:
In this case I believe that this is written in Java for speed, so we can't use `collect` here without converting it to Enso.